src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -134,11 +173,7 @@ protected Builder(ConnectionSettings connectionSettings) {
      */
     protected Builder(ServiceApiSettings settings) {
       this();
-      if (settings.connectionSettings != null) {
-        channelProvider = createChannelProvider(settings.connectionSettings);
-      } else {
-        channelProvider = createChannelProvider(settings.channel, settings.shouldAutoCloseChannel);
-      }
+      this.channelProvider = settings.channelProvider;
```

Comment:
Copying channelProvider here will result reusing the same channel instead of creating a new one.